honeybee_schema/radiance/material.py

```diff
@@ -0,0 +1,280 @@
+                    'not realistic (default: 0).'
+    )
+
+
+class Trans(Plastic):
+    """Radiance Translucent material."""
+
+    type: constr(regex='^Trans$') = 'Trans'
+
+    transmitted_diff: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is transmitted diffusely in '
+                    'a scattering fashion (default: 0).'
+    )
+
+    transmitted_spec: float = Field(
+        default=0,
+        description='The fraction of transmitted light that is not diffusely scattered '
+                    '(default: 0).'
+    )
+
+
+class Glass(ModifierBase):
+    """Radiance glass material."""
+
+    type: constr(regex='^Glass$') = 'Glass'
+
+    r_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the red channel transmissivity '
+                    '(default: 0).'
+    )
+
+    g_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the green channel transmissivity '
+                    '(default: 0).'
+    )
+
+    b_transmissivity: float = Field(
+        default=0.0,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the blue channel transmissivity '
+                    '(default: 0).'
+    )
+
+    refraction_index: float = Field(
+        default=1.52,
+        ge=0,
+        le=1,
+        description='A value between 0 and 1 for the index of refraction '
+                    '(default: 1.52).'
+    )
+
+
+class BSDF(ModifierBase):
+    """Radiance BSDF (Bidirectional Scattering Distribution Function) material."""
+
+    type: constr(regex='^BSDF$') = 'BSDF'
+
+    up_orientation: List[float] = Field(
+        default=(0.01, 0.01, 1.00),
+        description='Vector as sequence that sets the hemisphere that the BSDF material '
+                    'faces. (default: (0.01, 0.01, 1.00).'
+    )
+
+    thickness: float = Field(
+        default=0,
+        description='Optional number to set the thickness of the BSDF material '
+                    'Sign of thickness indicates whether proxied geometry is '
+                    'behind the BSDF surface (when thickness is positive) or in '
+                    'front (when thickness is negative)(default: 0).'
+    )
+
+    function_file: str = Field(
+        default='.',
+        min_length=1,
+        max_length=100,
+        description='Optional input for function file (default: ".").'
+    )
+
+    transform: str = Field(
+        default=None,
+        min_length=1,
+        max_length=100,
+        description='Optional transform input to scale the thickness and reorient '
+                    'the up vector (default: None).'
+    )
+
+    bsdf_data: bytes = Field(
+        ...,
+        description='BSDF xml file data as bytes.'
+    )
+
+    front_diffuse_reflectance: List[float] = Field(
+        default=None,
```

Comment:
list length check is missing for all the fields here. I just leave the comment for this one.

Reply:
You also probably should add a `validator` to ensure that each of the three values is between 0 and 1.

Reply:
As explained above, the reason I left this blank was because I couldn't find a Field arg that checked length, But I can do it with min/max length of 3.